sdk/containerregistry/azure-containerregistry/azure/containerregistry/_container_registry_client.py

```diff
@@ -764,3 +760,188 @@ def update_repository_properties(self, *args, **kwargs):
+        return DownloadManifestResult(digest=digest, data=content_stream, manifest=manifest)
+
+    @distributed_trace
+    def download_blob(self, repository, digest, **kwargs):
+        # type: (str, str, **Any) -> DownloadBlobResult | None
+        """Download a blob that is part of an artifact.
+
+        :param str repository: Name of the repository
+        :param str digest: The digest of the blob to download.
+        :returns: DownloadBlobResult or None
+        :rtype: ~azure.containerregistry.DownloadBlobResult or None
+        :raises ValueError: If the parameter repository or digest is None.
+        """
+        if repository is None or digest is None:
+            raise ValueError("The parameter repository and digest cannot be None.")
+
+        result = self._client.container_registry_blob.get_blob(repository, digest, **kwargs)
+        
+        if not result:
+            return result
+        
+        data = BytesIO(result.response.internal_response.content)
+        digest = _compute_digest(data)
+        return DownloadBlobResult(data=data, digest=digest)
+            
+
+    @distributed_trace
+    def delete_manifest(self, repository, tag_or_digest, **kwargs):
+        # type: (str, str, **Any) -> None
+        """Delete a manifest. If the manifest cannot be found or a response status code of
+        404 is returned an error will not be raised.
+
+        :param str repository: Name of the repository the manifest belongs to
+        :param str tag_or_digest: Tag or digest of the manifest to be deleted
+        :returns: The Http response from the service.
+        :rtype: ~azure.core.pipeline.transport._requests_basic.RequestsTransportResponse
+        :raises: ~azure.core.exceptions.HttpResponseError
+
+        Example
+
+        .. code-block:: python
+
+            from azure.containerregistry import ContainerRegistryClient
+            from azure.identity import DefaultAzureCredential
+            endpoint = os.environ["CONTAINERREGISTRY_ENDPOINT"]
+            client = ContainerRegistryClient(endpoint, DefaultAzureCredential(), audience="my_audience")
+            client.delete_manifest("my_repository", "my_tag_or_digest")
+        """
+        if _is_tag(tag_or_digest):
+            tag_or_digest = self._get_digest_from_tag(repository, tag_or_digest)
+
+        response = self._client.container_registry.delete_manifest(
+            repository, tag_or_digest, cls=_return_response, **kwargs
+        )
+        return response.http_response
+
+    @distributed_trace
+    def delete_blob(self, repository, tag_or_digest, **kwargs):
+        # type: (str, str, **Any) -> IO
+        """Delete a blob. If the blob cannot be found or a response status code of
+        404 is returned an error will not be raised.
+
+        :param str repository: Name of the repository the manifest belongs to
+        :param str tag_or_digest: Tag or digest of the blob to be deleted
+        :returns: IO
+        :rtype: IO
```

Comment:
Why would delete_blob return an IO?

Reply:
It returns from the call in generated call: https://github.com/Azure/azure-sdk-for-python/blob/d265b84c9cac5628b41dd29a0d001a37c9e7d1f5/sdk/containerregistry/azure-containerregistry/azure/containerregistry/_generated/operations/_container_registry_blob_operations.py#L545

Reply:
Delete shouldn't return anything - so this seems like a bug in the swagger?

Reply:
I'm not sure, [delete manifest](https://github.com/Azure/azure-rest-api-specs/blob/c8d9a26a2857828e095903efa72512cf3a76c15d/specification/containerregistry/data-plane/Azure.ContainerRegistry/stable/2021-07-01/containerregistry.json#L178) and [delete blob](https://github.com/Azure/azure-rest-api-specs/blob/c8d9a26a2857828e095903efa72512cf3a76c15d/specification/containerregistry/data-plane/Azure.ContainerRegistry/stable/2021-07-01/containerregistry.json#L336) both have return in swagger.

Reply:
This is probably one to consult with the feature crew/service team on.
Normally a delete API wouldn't return anything, because if the result isn't returned successfully (e.g. intermittent network failure), there's no way to retry and the response data is lost.
It would be good to confirm with the service team/feature crew:
- Does the service actually return a response here or is it a bug in the swagger spec
- If so, what is this data and how is a customer expected to use it?

To unblock this PR, I would just return None from a delete for now.

Reply:
From what I can recall from earlier reviews, these APIs _should_ not have returned anything from the server (for exactly the reasons stated above). @annatisch's suggestion is the right way to go.

Reply:
Thanks for the feedback! Have changed to return None here, and have sent out the email to the contact in service team.